dashboard/utils.py

```diff
@@ -142,6 +142,27 @@ def is_enrolled_mmtrack(self, course_id):
         else:
             return course_id in self.paid_course_ids
 
+    def paid_but_not_enrolled_mmtrack(self, course_id):
```

Comment:
- Can you make a function `has_paid` which just returns `True` or `False` depending on if they paid? You should extract the logic from `is_enrolled_mmtrack` which checks both FA and non-FA cases.
 - Then change `is_enrolled_mmtrack` to use this function `has_paid`
 - Then you can change this function to `return self.has_paid(course_id) and not self.is_enrolled(course_id)`